.github/workflows/install-with-rancher.yaml

```diff
@@ -0,0 +1,81 @@
+name: Install ROS on a server through Rancher
+
+on:
+  workflow_dispatch:
+    inputs:
+      registry:
+        required: false
+        default: quay.io
+        type: string
+        description: Registry to use
+      org:
+        required: false
+        default: costoolkit
+        type: string
+        description: Organization to use
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_USERNAME: ${{ secrets.QUAY_USERNAME }}
+      DOCKER_PASSWORD: ${{ secrets.QUAY_PASSWORD }}
+      DOCKER_REGISTRY: ${{ github.event.inputs.registry }}
+      REPO: ${{ github.event.inputs.registry }}/${{ github.event.inputs.org }}/os2-ci
+      REPO_ROS_OPERATOR: ${{ github.event.inputs.registry }}/${{ github.event.inputs.org }}/ros-operator-ci
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Make images (container/iso/qcow2/etc.)
+        run: make ci
+      - name: Upload results
+        uses: actions/upload-artifact@v3
+        with:
+          name: artifacts
+          path: dist
+          if-no-files-found: error
+      - name: Release space from worker ♻
+        if: always()
+        run: |
+          sudo rm -rf build bin dist
+          docker system prune -f -a --volumes
+  install-with-rancher:
+    runs-on: self-hosted
+    needs: build
+    container:
+      image: opensuse/leap:latest
+      env:
+        CLUSTER_NAME: cluster-k3s
+        CLUSTER_NS: fleet-default
+        K8S_VERSION: v1.21.9+k3s1
+        KUBECONFIG: /etc/rancher/k3s/k3s.yaml
+      options: --privileged
+    steps:
+      - name: Install dependencies
+        run: zypper -n in -l qemu-kvm libvirt virt-install sudo curl helm git-core tar
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: artifacts
+      - name: Install Rancher
+        run: scripts/install-rancher
+      - name: Configure Rancher to provision RancherOS on a test server
+        run: scripts/configure-rancher-step-01
+      - name: Install RancherOS on the test server
+        run: script -e -c scripts/install-ros-node
+      - name: Configure Rancher to add the test server in a test cluster
+        run: scripts/configure-rancher-step-02
+      - name: Add the test server in the test cluster
+        run: scripts/add-node-in-cluster
```

Comment:
The more I look at it as this looks like our first e2e test! can we do that more explicit and group this somewhere (makefile, a common bash entrypoint, ...) that runs it programmatically?

Reply:
I can group this yes, but in that case I will have less steps in the CI and this could be more complicated to debug in case of issue. Or I can split in different steps in a Makefile and add a step to run all at once it needed? I will check this.

Reply:
So I can either keep the same kind of scheduling like this:
```
      - name: Install Rancher
        run: cd tests && make install-rancher
      - name: Configure Rancher to provision RancherOS on a test server
        run: cd tests && make configure-rancher-step-01
      - name: Install RancherOS on the test server
        run: cd tests && make install-ros-node
      - name: Configure Rancher to add the test server in a test cluster
        run: cd tests && make configure-rancher-step-02
      - name: Add the test server in the test cluster
        run: cd tests && make add-node-in-cluster
```
Or replace the whole stack by something like this:
```
      - name: e2e - bootstrapping node with Rancher
        run: cd tests && make e2e-bootstraping-node-with-rancher
```

What do you prefer?

Reply:
Test with scripts moved in `tests/` and added in the `Makefile`: https://github.com/ldevulder/os2/actions/runs/2014938295